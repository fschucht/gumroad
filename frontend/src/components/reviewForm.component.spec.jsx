import renderer from 'react-test-renderer';
import { ReviewForm } from './reviewForm.component';

describe('ReviewForm', () => {
  beforeEach(() => jest.resetAllMocks())

  describe('when the user has already reviewed', () => {
    it('should return null', () => {
      jest.spyOn(Storage.prototype, 'getItem').mockReturnValueOnce(undefined)

      const renderedComponent = renderer.create(<ReviewForm hasReviewed />).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })

  describe('when the user has not reviewed', () => {
    it('should return null', () => {
      jest.spyOn(Storage.prototype, 'getItem').mockReturnValueOnce('api-token')

      const renderedComponent = renderer.create(<ReviewForm hasReviewed={false} />).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })

  describe('when there is no api token in the localstorage', () => {
    it('should render a login message', () => {
      jest.spyOn(Storage.prototype, 'getItem').mockReturnValueOnce(undefined)

      const renderedComponent = renderer.create(<ReviewForm hasReviewed={false} />).toJSON()

      expect(renderedComponent).toMatchSnapshot()
    })
  })
})