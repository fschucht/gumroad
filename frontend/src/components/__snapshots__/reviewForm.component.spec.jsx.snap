// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm when the user has already reviewed should return null 1`] = `null`;

exports[`ReviewForm when the user has not reviewed should return null 1`] = `
<div
  className="view"
>
  <form
    className="form"
  >
    <div
      className="rating-input"
    >
      <p
        className="text"
      >
        Your Rating
      </p>
      <div
        className="rating-input__stars"
      >
        <svg
          className="rating-input__star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          star.svg
        </svg>
        <svg
          className="rating-input__star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          star.svg
        </svg>
        <svg
          className="rating-input__star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          star.svg
        </svg>
        <svg
          className="rating-input__star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          star.svg
        </svg>
        <svg
          className="rating-input__star"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          star.svg
        </svg>
      </div>
    </div>
    <textarea
      className="textarea"
      onChange={[Function]}
      placeholder="Your Comment"
      value=""
    />
  </form>
  <button
    className="button button--primary"
    disabled={true}
    onClick={[Function]}
  >
    Submit Review
  </button>
</div>
`;

exports[`ReviewForm when there is no api token in the localstorage should render a login message 1`] = `
<div
  className="view"
>
  <p
    className="text"
  >
    Login in to review this product.
  </p>
</div>
`;
